backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py

```diff
@@ -348,63 +376,92 @@ def _should_index_depot_file(self, content: ContentInContext) -> bool:
 
         return True
 
+    @classmethod
+    def create_es_range(cls, range_from, range_to):
+        # simple date/number facet (no histogram)
+        if range_from and range_to:
+            return {"gte": range_from, "lte": range_to}
+
+        if range_from:
+            return {"gte", range_from}
+
+        if range_to:
+            return {"lte", range_to}
+
+        return None
+
     def search_content(
         self,
-        search_string: str,
-        size: typing.Optional[int],
-        page_nb: typing.Optional[int],
-        content_types: typing.Optional[typing.List[str]] = None,
-        show_deleted: bool = False,
-        show_archived: bool = False,
-        show_active: bool = True,
+        simple_parameters: ContentSearchFilterQuerySchema,
+        advanced_parameters: typing.Optional[AdvancedContentSearchParameters] = None,
     ) -> ContentSearchResponse:
         """
         Search content into elastic search server:
         - do no show archived/deleted content by default
         - filter content found according to workspace of current_user
         """
-        if not search_string:
+        if not simple_parameters.search_string:
             return EmptyContentSearchResponse()
         filtered_workspace_ids = self._get_user_workspaces_id(min_role=UserRoleInWorkspace.READER)
         # INFO - G.M - 2019-05-31 - search using simple_query_string, which means user-friendly
         # syntax to match complex case,
         # see https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html
+        es_search_fields = []
+
+        search_fields = (
+            advanced_parameters.search_fields
+            if advanced_parameters and advanced_parameters.search_fields
+            else ["label", "raw_content", "comments"]
+        )
+
+        if "label" in search_fields:
+            # TODO - G.M - 2021-02-08 -  we may want to split exact and not exact search to allow
+            # doing exact search efficiently.
+            es_search_fields.extend(
+                ["label.exact^8", "label^5", "filename.exact", "filename", "file_extension"]
+            )
+
+        if "raw_content" in search_fields:
+            es_search_fields.extend(
+                [
+                    "raw_content.exact^3",
```

Comment:
The boost values have been changed here, will this change the results ordering?

Reply:
This is not intentional (I guess? @inkhey?). I fail to spot the ones that changed.